test/ngMock/angular-mocksSpec.js

```diff
@@ -1943,10 +1943,30 @@ describe('ngMock', function() {
         );
         they('should ignore query param when matching in ' + routeShortcut + ' $prop method', methods,
           function() {
-            hb[routeShortcut](this, '/route/:id').respond('path');
+            var paramsSpy = jasmine.createSpy('paramsSpy');
+            hb[routeShortcut](this, '/route/:id').respond(function(method, url, data, headers, params) {
+              paramsSpy(params);
+              // status, response, headers, statusText, xhrStatus
+              return [200, 'path', { 'x-header': 'foo' }, 'OK', 'complete'];
```

Comment:
I don't think `complete` is used (here and below).

Reply:
I'm verifying that `params` are correct. Without the fix, `paramSpy` is called with `{id: '123?q=str&foo=bar', q: 'str', foo: 'bar'}`.